### PR TITLE
fix preprocess script with processbar, integrity check and logs

### PR DESCRIPTION
**Pull request by lidanqing-vv**
Fix reviews of PR [16529](https://github.com/PaddlePaddle/Paddle/pull/16529), rebased on release 1.4

1. process bar added
2. add check size and check integrity for existing binary file, for comment: [Exisiting file is incomplete](https://github.com/PaddlePaddle/Paddle/pull/16529#discussion_r270291156) 
3. Write clear logging
4. The unzipped file is in .cache/paddle/dataset/int8/download/full_data, for comment: [unzip not inplace](https://github.com/PaddlePaddle/Paddle/pull/16529#discussion_r270301436 )

### CODE REVIEW

**Comment by lidanqing-vv** [comment]
@kbinias @bingyanghuang @Sand3r- @luotao1  Please review this new PR rebased on release 1.4. (old one closed). Thank you

File: paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py [hunk 1]
```diff
@@ -93,70 +90,140 @@ def download_unzip():
...
+    print('Start to download and convert 50000 images to binary file...')
+    cache_folder = os.path.expanduser('~/.cache/paddle/dataset/int8/download')
+    extract_folder = os.path.join(cache_folder, 'full_data')
+    data_dir = os.path.join(extract_folder, DATA_DIR_NAME)
+    file_list = os.path.join(data_dir, 'val_list.txt')
+    zip_path = os.path.join(cache_folder, 'full_imagenet_val.tar.gz')
+    output_file = os.path.join(cache_folder, 'int8_full_val.bin')
+    retry = 0
+    try_limit = 3
+    target_hash = '8dc592db6dcc8d521e4d5ba9da5ca7d2'
```
Comment (kbinias) [inline paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py]: Change name to `TARGET_HASH` and move to the top of the file, e.g. below `IMG_DIR_NAME = 'var'`

File: paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py [hunk 2]
```diff
@@ -70,15 +75,7 @@ def process_image(img_path, mode, color_jitter, rotate):
...
-def download_unzip():
-    int8_download = 'int8/download'
-
-    target_name = 'data'
-
-    cache_folder = os.path.expanduser('~/.cache/paddle/dataset/' +
-                                      int8_download)
-
-    target_folder = os.path.join(cache_folder, target_name)
+def download_concat(cache_folder, zip_path):
```
Comment (kbinias) [inline paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py]: Remove empty line

File: paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py [hunk 3]
```diff
@@ -93,70 +90,140 @@ def download_unzip():
...
+        return True
+    else:
+        return False
 
 
-def reader():
-    data_dir = download_unzip()
-    file_list = os.path.join(data_dir, 'val_list.txt')
-    output_file = os.path.join(data_dir, 'int8_full_val.bin')
+def converter(file_list, data_dir, output_file):
```
Comment (kbinias) [inline paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py]: Change name `converter` to `convert` to be consistent with the name `run_convert()`

File: paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py [hunk 4]
```diff
@@ -30,6 +31,11 @@
 
 SIZE_FLOAT32 = 4
 SIZE_INT64 = 8
+FULL_SIZE_BYTES = 30106000008
+FULL_IMAGES = 50000
+DATA_DIR_NAME = 'ILSVRC2012'
+IMG_DIR_NAME = 'var'
+TARGET_HASH = '8dc592db6dcc8d521e4d5ba9da5ca7d2'
```
Comment (lidanqing-vv) [inline paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py:37]: Test review. Please ignore this comment.

**Review by kbinias — COMMENTED** [review]
I checked this code on my developer machine, the expected result is binary file.

**Comment by luotao1** [comment]
The log is very nice, thanks very much!
```
Start to download and convert 50000 images to binary file...
Downloading full ImageNet Validation dataset ...

Downloaded part /dev/shm/.cache/paddle/dataset/int8/download/ILSVRC2012_img_val.tar.gz.partaa


Downloaded part /dev/shm/.cache/paddle/dataset/int8/download/ILSVRC2012_img_val.tar.gz.partab

Extracting...

Extracted. Full Imagenet Validation dataset is located at /dev/shm/.cache/paddle/dataset/int8/download/full_data/ILSVRC2012

Converting 50000 images to binary file ...

[===========                                       ]22%
```

**Review by luotao1 — APPROVED** [review]
Please cherry-pick to release/1.4, and note that the commit info should have `test=release/1.4`.

**Pull request merged**